### PR TITLE
made sda and scl public for i2c interfaces

### LINKED ISSUE #150: No way to take ownership of the individual i2c pins without losing their type
Currently `I2CExternalPins` and `I2CInternalPins` have private fields, and there is no `into_inner()` or similar. As far as I can tell, this means that it is impossible to take ownership of these pins without losing their specific pin type. The type would be nice when moving these pins to a library crate that doesn't depend on `microbit` but does want to know which microbit I2C interface it is working with.

As far as I can tell, there's no reason not to make these fields public. I will submit a PR.

### PR DESCRIPTION
**Pull request by BartMassey**
Made the `sda` and `scl` pins in the various `I2c` `Pins` struct public.

Closes #150

### CODE REVIEW

**Comment by qwandor** [comment]
Please add this to the changelog.

**Review by qwandor — APPROVED** [review]

**Comment by BartMassey** [comment]
Done. Also fixed a screwup. Apologies.

Will fix the MSRV break in a moment.

**Comment by BartMassey** [comment]
Nope. Fix did not work. Help figuring this out appreciated. MSRV workflow exits with status 101, no obvious log message I could see.

**Comment by BartMassey** [comment]
Looks like the version of `fixed` used by `nrf51-hal` requires Rust 1.78.0 even on feature `embedded-hal-02`. Bummer.

Shall we update the MSRV of this, or shall I try to get a fix in upstream?

**Comment by qwandor** [comment]
Weird, what changed to break that? Personally I don't mind updating the MSRV, but I'd like to understand what caused it to break.

**Comment by BartMassey** [comment]
Looks like the version of `fixed` used in `nrf-hal` got bumped from 1.0.0 to 1.25.1 on 2024-03-06 when `nrf-hal` was being prepared for 0.18 release. The new `fixed` has an MSRV.

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Set MSRV for `microbit` and `microbit-v2` crates.
 - Bumped MSRV to 1.79.0.
 
+## [0.15.1] - 2024-07-30
```
Comment (qwandor) [inline CHANGELOG.md:13]: This should include the items under the 'Unreleased' heading above.

Comment (BartMassey) [reply CHANGELOG.md]: Good catch. Fixed, thanks much.

**Review by qwandor — APPROVED** [review]

**Pull request merged**